pointcloud/point.go

```diff
@@ -126,11 +126,39 @@ func (bp *basicData) Value() int {
 	return bp.value
 }
 
-func (bp *basicData) SetIntensity(v uint16) Data {
-	bp.intensity = v
-	return bp
-}
+// MarshalBinary checks for the presence of color and value data. If present, it will store said data in an order byte
+// array.
+func (bp *basicData) MarshalBinary() (data []byte, err error) {
+	var dataBytes []byte
```

Comment:
give this object capacity equal to with the number of max number of potential bytes  i.e.
`dataBytes := make([]bytes, 0, 8)`

Reply:
If say the metadata header states that color data is not present, then we'd want MarshalBinary to return a smaller serialized []byte as to avoid sending excess data that is just 0s. Are you suggesting we always want to have a consistent data return for `MarshalBinary`, independent on the data that may or not be present in it?

Reply:
capacity is not length. in Go, every time you write to your slice, if you exceed the capacity of the underlying array, then a new array is allocated with double the capacity, the data is copied, and the pointer switches over to the new underlying array. The old underlying array is then garbage collected if there are no references to that array. So extra zeros will not be copied. The slice begins with length 0, and only grows as you append - setting the capacity to 8 just reduces the load of underlying copies and garage collection

Reply:
https://dev.to/jonathanlawhh/golang-slice-vs-array-and-should-i-declare-my-slice-size-4kng